library/azure_rm_virtualmachinescalesetextension.py

```diff
@@ -0,0 +1,269 @@
+    type_handler_version:
+        description:
+            - The type version of the extension handler.
+    settings:
+        description:
+            - Json formatted public settings for the extension.
+    protected_settings:
+        description:
+            - Json formatted protected settings for the extension.
+    auto_upgrade_minor_version:
+        description:
+            - Whether the extension handler should be automatically upgraded across minor versions.
+        type: bool
+    state:
+        description:
+            - Assert the state of the SQL server. Use 'present' to create or update a server and
+              'absent' to delete a server.
+        default: present
+        choices:
+            - absent
+            - present
+
+
+extends_documentation_fragment:
+    - azure
+
+author:
+    - "Zim Kalinowski (@zikalino)"
+'''
+
+EXAMPLES = '''
+    - name: Create VMSS Extension
+      azure_rm_virtualmachinescalesetextension:
+        name: myvmssextension
+        location: eastus
+        resource_group: Testing
+        vmss_name: myvm
+        publisher: Microsoft.Azure.Extensions
+        type: CustomScript
+        type_handler_version: 2.0
+        settings: '{"commandToExecute": "hostname"}'
+        auto_upgrade_minor_version: true
+
+    - name: Delete VMSS Extension
+      azure_rm_virtualmachinescalesetextension:
+        name: myvmssextension
+        location: eastus
+        resource_group: Testing
+        vmss_name: myvm
+        state: absent
+'''
+
+RETURN = '''
+id:
+    description:
+        - Instance resource ID
+    returned: always
+    type: str
+    sample: /subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/TestGroup/providers/Microsoft.Compute/scalesets/myscaleset/extensions/myext
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMVMSSExtension(AzureRMModuleBase):
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            ),
+            location=dict(
+                type='str'
+            ),
+            vmss_name=dict(
+                type='str'
+            ),
+            publisher=dict(
+                type='str'
+            ),
+            type=dict(
+                type='str'
+            ),
+            type_handler_version=dict(
+                type='str'
+            ),
+            auto_upgrade_minor_version=dict(
+                type='bool'
+            ),
+            settings=dict(
```

Comment:
pls add spec for dict type

Reply:
this depends on the extension type so can't be specified here.

Reply:
can you add above description in doc part, it's hard for user to fill a dict without clear description or sample.

Reply:
I have added descriprion and url to more info